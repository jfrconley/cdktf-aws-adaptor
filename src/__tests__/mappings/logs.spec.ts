import { CloudwatchLogGroup } from "@cdktf/provider-aws/lib/cloudwatch-log-group/index.js";
import { CfnLogGroup } from "aws-cdk-lib/aws-logs";
import { describe } from "vitest";
import { itShouldMapCfnElementToTerraformResource } from "../helpers.js";

describe("Logs mappings", () => {
  itShouldMapCfnElementToTerraformResource(
    CfnLogGroup,
    {
      kmsKeyId: "test-kms-key-id",
      logGroupName: "test-log-group-name",
      retentionInDays: 1,
      tags: [{
        key: "test-tag-key",
        value: "test-tag-value",
      }],
    },
    CloudwatchLogGroup,
    {
      tags: {
        "test-tag-key": "test-tag-value",
      },
      name: "test-log-group-name",
      retentionInDays: 1,
      kmsKeyId: "test-kms-key-id",
    },
  );
});
