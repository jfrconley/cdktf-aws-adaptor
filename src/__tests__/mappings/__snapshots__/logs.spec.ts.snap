// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logs mappings > Should map AWS::Logs::LogGroup 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudwatch_log_group\\": {
      \\"resource_22C949BF\\": {
        \\"kms_key_id\\": \\"test-kms-key-id\\",
        \\"name\\": \\"test-log-group-name\\",
        \\"retention_in_days\\": 1,
        \\"tags\\": {
          \\"test-tag-key\\": \\"test-tag-value\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.22.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_cloudwatch_log_group.resource_22C949BF.arn}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
